src/OSPSuite.R/Services/ConcurrentSimulationRunner.cs

```diff
@@ -13,6 +13,37 @@
 
 namespace OSPSuite.R.Services
 {
+   public class SimulationWithBatchOptions
+   {
+      public IModelCoreSimulation Simulation { get; set; }
+      public List<SimulationBatchRunValues> SimulationBatchRunValues { get; } = new List<SimulationBatchRunValues>();
+      public SimulationBatchOptions SimulationBatchOptions { get; set; }
+      public void AddSimulationBatchRunValues(SimulationBatchRunValues simulationBatchRunValues)
+      {
+         SimulationBatchRunValues.Add(simulationBatchRunValues);
+      }
+   }
+
+   class SimulationBatchRunOptions
```

Comment:
I am a bit confused where
We have 
```
SimulationBatchOptions
SimulationBatchRunValues
SimulationBatchRunOptions
SimulationBatch
SimulationWithBatchOptions
```

is this maybe too much ?

Reply:
SimulationBatchRunOptions is internal. It is needed because this is in the signature of the runSimulationBatch method. This method needs all the information on it but only accepts a generic data type in order to be used by the ConcurrencyManager.RunAsync method.

SimulationWithBatchOptions used to prepare the batches from the R side. This is the only new public class which is used for handling the information set from R.

SimulationBatchRunValues, SimulationBatch already existing on SimulationBatch